lib/langchain/llm/openai.rb

```diff
@@ -18,10 +18,12 @@ class OpenAI < Base
       dimension: 1536
     }.freeze
     LENGTH_VALIDATOR = Langchain::Utils::TokenLength::OpenAIValidator
-    ROLE_MAPPING = {
-      "ai" => "assistant",
-      "human" => "user"
-    }
+    COMPLETION_RESPONSE_PATH = ["choices", 0, "message"]
+    STREAMING_RESPONSE_PATH = ["choices", 0, "delta"]
+    ROLE_KEY = :role
+    USER_ROLE_KEY = "user"
+    AI_ROLE_KEY = "assistant"
+    SYSTEM_ROLE_KEY = "system"
```

Comment:
this is starting to get strange. The problem is the conflict between:
- We don't want the LLMs to know about anything else
- We can't put code in other places like `Message` or `Conversation` because when adding more LLMs we'll forget about those

Reply:
> We can't put code in other places like Message or Conversation because when adding more LLMs we'll forget about those

This one is not a valid issue. If this ever happens -- it just means that our test coverage is bad.

Reply:
agree.. then maybe we just need to accept having the Conversation know about each LLM specifics, instead of having each LLM having to provide things in this way (indirectly knowing that someone needs these things to be defined).

And with that we go back to the solution where Message (or could be Conversation or another class in the same "domain") knows how to parse an OpenAI result, Google Palm 2 result, etc